tests/eth_provider.rs

```diff
@@ -386,7 +385,7 @@ async fn test_to_starknet_block_id(#[future] katana: Katana, _setup: ()) {
 
     // Then
     assert_eq!(pending_starknet_block_id, starknet::core::types::BlockId::Number(0x1234_u64));
-    assert_eq!(some_starknet_block_hash, starknet::core::types::BlockId::Hash(FieldElement::from(0x1234_u64)));
+    assert_eq!(some_starknet_block_hash, starknet::core::types::BlockId::Number(0x1234_u64));
```

Comment:
why is a block_hash equal to a block number

Reply:
So when you convert to a starknet block id using `to_starknet_block_id`, and the input is a hash, you actually can't return a Starknet hash, because the eth hash format doesn't fit in the FieldElement :/ so I convert it to a block number by looking it up in the DB

Reply:
specifically in the hive context?

Reply:
Mmmmm I guess it's only in hive yes, because you probably won't query a B256 hash in another context

Reply:
can we rename the vars? its weird to see some_starknet_block_hash == starknet::core::types::BlockId::Number(0x1234_u64)

Reply:
maybe that's better yeah

Reply:
ok will do

Reply:
done